openstack/compute/v2/extensions/keypairs/urls.go

```diff
@@ -16,10 +16,14 @@ func createURL(c *gophercloud.ServiceClient) string {
 	return resourceURL(c)
 }
 
-func getURL(c *gophercloud.ServiceClient, name string) string {
-	return c.ServiceURL(resourcePath, name)
+func getURL(c *gophercloud.ServiceClient, name string, userID string) string {
```

Comment:
By using a `GetOpts` struct, you should be able to revert `getURL` back to the original code and just append the query params.